### PR TITLE
#5142 Migrate google mock endpoint to `ConfigurationProvider`

### LINKED ISSUE #5142: Rewrite `Google` mock to use `ConfigurationProvider`

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR migrated google mock endpoint to `ConfigurationProvider`

close #5142 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: test/source/mock/google/google-endpoints.ts [hunk 1]
```diff
@@ -138,76 +150,52 @@ export const getMockGoogleEndpoints = (oauth: OauthMock): HandlersDefinition =>
...
+        isDefault: true,
+        isPrimary: true,
+        treatAsAlias: false,
+        verificationStatus: 'accepted',
+      };
+      // If the account is a compatibility account, return specific aliases.
+      // This account is used in hundreds of tests, so we handle it separately to avoid duplicating the code.
+      if (acct === 'flowcrypt.compatibility@gmail.com') {
+        const alias = 'flowcryptcompatibility@gmail.com';
+        return {
```
Comment (ioanmo226) [inline test/source/mock/google/google-endpoints.ts:172]: @sosnovsky As you can see, when acct is `flowcrypt.compatibility@gmail.com`, I returned specific HARD_CODED aliases.
Do you agree with this or want me to duplicate below same configuration to every compatibility tests?

Comment (sosnovsky) [reply test/source/mock/google/google-endpoints.ts:172]: It's ok to have it hardcoded here, no need for duplicated code

File: test/source/mock/google/google-endpoints.ts [hunk 2]
```diff
@@ -81,55 +120,28 @@ export const getMockGoogleEndpoints = (oauth: OauthMock): HandlersDefinition =>
...
-              { person: { emailAddresses: [{ metadata: { primary: true }, value: 'testsearchorder6@flowcrypt.com' }] } },
-            ],
-          };
-        } else {
-          return empty;
-        }
-      } else {
-        return empty;
+      if (!config.contacts) {
+        return {};
```
Comment (sosnovsky) [inline test/source/mock/google/google-endpoints.ts:127]: What if we'll return `{}` even when config isn't set? (and for `/othercontacts` too)
Then we won't need to declare empty config like we have in some tests `{ google: { contacts: [] } }`

**Review by sosnovsky — COMMENTED** [review]
Well done 👍

**Review by sosnovsky — APPROVED** [review]
Nice 👍

**Pull request merged**